### PR TITLE
CB-5563 [TE] Change credentials - 'save credentials' is not marked for auth even if it is saved

### PR DESCRIPTION
**Pull request by sergeyteleshev**

### CODE REVIEW

**Review by Wroud — APPROVED** [review]

File: webapp/packages/plugin-connections/src/DatabaseAuthDialog/DatabaseCredentialsAuthDialog/useDatabaseCredentialsAuthDialog.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export function useDatabaseCredentialsAuthDialog(
           for (const id of this.networkHandlers) {
             const handler = connection.networkHandlersConfig.find(handler => handler.id === id);
 
-            if (handler && (handler.userName || handler.authType !== NetworkHandlerAuthType.Password)) {
```
Comment (devnaumov) [inline webapp/packages/plugin-connections/src/DatabaseAuthDialog/DatabaseCredentialsAuthDialog/useDatabaseCredentialsAuthDialog.ts]: What changed? Why don't we want this check anymore?

Comment (sergeyteleshev) [reply webapp/packages/plugin-connections/src/DatabaseAuthDialog/DatabaseCredentialsAuthDialog/useDatabaseCredentialsAuthDialog.ts:99]: Cause of this case:
https://dbeaver.atlassian.net/browse/CB-5563?focusedCommentId=23902

Comment (sergeyteleshev) [reply webapp/packages/plugin-connections/src/DatabaseAuthDialog/DatabaseCredentialsAuthDialog/useDatabaseCredentialsAuthDialog.ts]: cause it does not apply for SSL (no `username` and `NetworkHandlerAuthType.Password`)
see the last attached case to the ticket in the comments

Comment (sergeyteleshev) [reply webapp/packages/plugin-connections/src/DatabaseAuthDialog/DatabaseCredentialsAuthDialog/useDatabaseCredentialsAuthDialog.ts]: reverted this logic. we need to groom this form in general cause it has several issues with SSH and SSL fields (created tickets for that)

**Pull request merged**